h2/src/main/org/h2/mvstore/CursorPos.java

```diff
@@ -8,7 +8,7 @@
 /**
  * A position in a cursor
  */
-public class CursorPos {
+public final class CursorPos {
```

Comment:
What do you want to achieve by making the class final?

Reply:
While I'm not a fan of code pollution with "final" for effectively immutable local variables either, but final classes is a slightly different story. Apart from preventing someone from unexpected sub-classing, it declares all method as final, and this fact is not statically known otherwise (unlike local vars), and it gives JIT much more freedom for call optimization, or even inlining. Where it materializes into something tangible in any specific case, that's a a different story, but it's such a little price to pay.

Reply:
The last time I did some benchmarking with final classes versus non-final classes, I couldn't find any difference. I read that the JVM analyzes at runtime which classes are effectively final (that is, have no class that extends it), and therefore it won't make a difference if you make the class final or not. That's why I wouldn't worry about making a class final.

If you do find that making it final improves performance, then that's fine! But it would be better to do that in a separate issue (to make this patch here smaller).

Reply:
What about: not use final in _this_ patch, and then you can create another issue that uses final for many important internal classes, where we discuss the pros and cons? Actually I don't have a strong preference, I just wasn't sure what your motivation is... I think using final to improve performance is fine, if there is some indication that it does...

Reply:
Ok, will do